src/test/java/org/jenkinsci/plugins/workflow/cps/DSLTest.java

```diff
@@ -420,6 +430,91 @@ public void namedSoleParamForStep() throws Exception {
+        InterpolatedSecretsAction reportAction = run.getAction(InterpolatedSecretsAction.class);
+        Assert.assertNotNull(reportAction);
+        Set<String> reportResults = reportAction.getResults();
+        MatcherAssert.assertThat(reportResults.size(), is(1));
+        MatcherAssert.assertThat(reportResults.iterator().next(), is("PASSWORD"));
+        LinearScanner scan = new LinearScanner();
+        FlowNode node = scan.findFirstMatch(run.getExecution().getCurrentHeads().get(0), new NodeStepTypePredicate(Functions.isWindows()? "bat" : "sh"));
+        ArgumentsAction argAction = node.getPersistentAction(ArgumentsAction.class);
+        Assert.assertFalse(argAction.isUnmodifiedArguments());
+        MatcherAssert.assertThat(argAction.getArguments().values().iterator().next(), instanceOf(ArgumentsAction.NotStoredReason.class));
+    }
+
+    @Test public void describableInterpolation() throws Exception {
+        final String credentialsId = "creds";
+        final String username = "bob";
+        final String password = "secr3t";
+        UsernamePasswordCredentialsImpl c = new UsernamePasswordCredentialsImpl(CredentialsScope.GLOBAL, credentialsId, "sample", username, password);
+        CredentialsProvider.lookupStores(r.jenkins).iterator().next().addCredentials(Domain.global(), c);
+        p.setDefinition(new CpsFlowDefinition(""
+                + "node {\n"
+                + "withCredentials([usernamePassword(credentialsId: 'creds', usernameVariable: 'USERNAME', passwordVariable: 'PASSWORD')]) {\n"
+                + "archiveArtifacts(\"${PASSWORD}\")"
+                + "}\n"
+                + "}", true));
+        WorkflowRun run = r.assertBuildStatus(Result.FAILURE, p.scheduleBuild2(0));
+        r.assertLogContains("Affected variable(s): [PASSWORD]", run);
+        InterpolatedSecretsAction reportAction = run.getAction(InterpolatedSecretsAction.class);
+        Assert.assertNotNull(reportAction);
+        Set<String> reportResults = reportAction.getResults();
+        MatcherAssert.assertThat(reportResults.size(), is(1));
+        MatcherAssert.assertThat(reportResults.iterator().next(), is("PASSWORD"));
+    }
+
+    @Test public void describableNoMetaStep() throws Exception {
+        final String credentialsId = "creds";
+        final String username = "bob";
+        final String password = "secr3t";
+        UsernamePasswordCredentialsImpl c = new UsernamePasswordCredentialsImpl(CredentialsScope.GLOBAL, credentialsId, "sample", username, password);
+        CredentialsProvider.lookupStores(r.jenkins).iterator().next().addCredentials(Domain.global(), c);
+        p.setDefinition(new CpsFlowDefinition(""
+                + "node {\n"
+                + "withCredentials([usernamePassword(credentialsId: 'creds', usernameVariable: 'USERNAME', passwordVariable: 'PASSWORD')]) {\n"
+                + "monomorphWithSymbolStep(monomorphSymbol([firstArg:\"${PASSWORD}\", secondArg:'two']))"
```

Comment:
This test triggered the following warning:
`WARNING	o.j.p.w.cps.CpsFlowExecution#maybeAutoPersistNode: Attempt to persist triggered IOException for node 7
java.lang.UnsupportedOperationException: Refusing to marshal java.util.HashMap$KeySet for security reasons; see https://jenkins.io/redirect/class-filter/`
and then further down:
`Caused: java.lang.RuntimeException: Failed to serialize org.jenkinsci.plugins.workflow.cps.actions.ArgumentsActionImpl#sensitiveVariables for class org.jenkinsci.plugins.workflow.cps.actions.ArgumentsActionImpl`